tests/sealing.rs

```diff
@@ -43,11 +43,21 @@ fn test_sealing_add() {
     let r2 = m0.seals().unwrap();
     assert_eq!(r2, a2);
 
+    // `SealFutureWrite` is new as of Linux 5.1, so be prepared for it to fail
+    // if we don't have it.
+    let future_write_seal = memfd::SealsHashSet::from_iter(vec![memfd::FileSeal::SealFutureWrite]);
```

Comment:
Doesn't this need to be guarded by `#[cfg(any(target_os = "android", target_os = "linux"))]`? Otherwise the compilation will fail due to the value being missing.

Reply:
Now that we're using rustix, sounds like reviving #15 would be possible…

Reply:
I've now added such a guard.

I expect FreeBSD support is possible, though it'll need a few additional changes outside the scope of this PR, so I'll submit a separate PR for it.

Reply:
Don't worry about it. Last time we tried to do it, we were blocked by the way the `libc` crate detects the version of FreeBSD, and we were ultimately blocked by it. I don't anticipate anything to have changed since.